pulp_ansible/app/galaxy/v3/serializers.py

```diff
@@ -333,3 +333,96 @@ class ClientConfigurationSerializer(serializers.Serializer):
     """Configuration settings for the ansible-galaxy client."""
 
     default_distribution_path = serializers.CharField(allow_null=True)
+
+
+
+class CollectionVersionSearchListSerializer(CollectionVersionListSerializer):
+
+    # broken ...
+    href = serializers.SerializerMethodField()
+
+    # override because RepositoryContent != CollectionVersion without a cast ...
+    pulp_id = serializers.SerializerMethodField()
+    namespace = serializers.SerializerMethodField()
+    name = serializers.SerializerMethodField()
+    version = serializers.SerializerMethodField()
+    is_highest = serializers.SerializerMethodField()
+    is_deprecated = serializers.SerializerMethodField()
+    created_at = serializers.SerializerMethodField()
+    updated_at = serializers.SerializerMethodField()
+    requires_ansible = serializers.SerializerMethodField()
+    dependencies = serializers.SerializerMethodField()
+    tags = serializers.SerializerMethodField()
+
+    repository_name = serializers.SerializerMethodField()
+
+    class Meta:
+        fields = (
+            "pulp_id",
+            "namespace",
+            "name",
+            "version",
+            "is_highest",
+            "is_deprecated",
+            "href",
+            "created_at",
+            "updated_at",
+            "requires_ansible",
+            "dependencies",
+            "repository_name",
+            "tags",
+        )
+        model = models.CollectionVersion
+
+    def get_collection_version(self, obj):
+        return obj.content.cast()
+
+    def get_pulp_id(self, obj):
+        return self.get_collection_version(obj).pulp_id
```

Comment:
Is `self.get_collection_version` cached? or calling it multiple times for each `SerializerMethodField` will trigger a new query?

Reply:
No caching yet. I still have to hack through how the obj is initially given to the serializer and the subsequent method calls.